dashboard/src/app/(authenticated)/[tenantId]/(diagram)/wfRun/[...ids]/components/ChildWorkflows.tsx

```diff
@@ -0,0 +1,144 @@
+import { RefreshCwIcon } from 'lucide-react'
+import { useParams, useSearchParams } from 'next/navigation'
+import { useEffect, useMemo, useState } from 'react'
+import useSWRInfinite, { SWRInfiniteKeyLoader } from 'swr/infinite'
+import { PaginatedWfRunIdList, searchWfRun } from '../../../wfSpec/[...props]/actions/searchWfRun'
+import { WfRunsHeader } from '../../../wfSpec/[...props]/components/WfRunsHeader'
+import { statusColors } from './Details'
+
+export default function ChildWorkflows({ parentWfRunId, spec }: { parentWfRunId: WfRunId; spec: WfSpec }) {
+  const tenantId = useParams().tenantId as string
+  const searchParams = useSearchParams()
+  const status = (searchParams.get('status') ? getStatus(searchParams.get('status')) || 'ALL' : 'ALL') as
+    | LHStatus
+    | 'ALL'
+  const [limit, setLimit] = useState<number>(SEARCH_DEFAULT_LIMIT)
+  const [resolvedWfRuns, setResolvedWfRuns] = useState<Record<string, WfRunResponse>>({})
+  const [window, setWindow] = useState<TimeRange>(TIME_RANGES[0])
+
+  const startTime = useMemo(() => {
+    if (window === -1) return undefined
+    const now = new Date()
+    const latestStart = now.toISOString()
+    const earliestStart = new Date(now.getTime() - window * 6e4).toISOString()
+
+    return {
+      latestStart,
+      earliestStart,
+    }
+  }, [window])
+
+  type StartTimeRange = { latestStart: string; earliestStart: string } | undefined
+  type ChildWfRunsKey = ['childWfRuns', LHStatus | 'ALL', string, number, StartTimeRange, string | undefined, WfRunId]
+
+  const getKey: SWRInfiniteKeyLoader<PaginatedWfRunIdList, ChildWfRunsKey | null> = (_pageIndex, previousPageData) => {
+    if (previousPageData && !previousPageData.bookmarkAsString) return null // reached the end
+    return [
+      'childWfRuns',
+      status,
+      tenantId,
+      limit,
+      startTime,
+      previousPageData?.bookmarkAsString,
+      parentWfRunId,
+    ] as ChildWfRunsKey
+  }
+  const { data, error, size, setSize } = useSWRInfinite<PaginatedWfRunIdList>(getKey, async (key: ChildWfRunsKey) => {
+    const [, status, tenantId, limit, startTime, bookmarkAsString, parentWfRunId] = key
+    return await searchWfRun({
+      wfSpecName: '',
+      status: status === 'ALL' ? undefined : status,
+      tenantId,
+      parentWfRunId,
+      limit,
+      bookmarkAsString,
+      variableFilters: [],
+      ...startTime,
+    })
+  })
+
+  const wfRunPromises = useMemo(() => {
+    return (
+      data?.flatMap(page =>
+        page.results.map(wfRunId => ({
+          wfRunId: wfRunId,
+          promise: getWfRun({ wfRunId, tenantId }),
+        }))
+      ) ?? []
+    )
+  }, [data, tenantId])
+
+  useEffect(() => {
+    wfRunPromises.forEach(async ({ wfRunId, promise }) => {
+      const data = await promise
+      setResolvedWfRuns(prev => ({
+        ...prev,
+        [wfRunId.id]: data,
+      }))
+    })
+  }, [wfRunPromises])
```

Comment:
This part should be a server side function and also resolve the promise in the server side.